tests/test_merge.py

```diff
@@ -26,3 +28,120 @@
 )
 def test_get_average_frame_count(samples: str, count: float) -> None:
     assert get_average_frame_count(samples) == count
+
+
+@pytest.mark.parametrize("insert_dso_name", [False, True])
+@pytest.mark.parametrize(
+    "stack, outcome_dict",
+    [
+        pytest.param(
+            "	 7f80 operator new+0x0 (/lib/libstdc++.so)",
+            dict(
+                dso_true="operator new (/lib/libstdc++.so)",
+                dso_false="operator new",
+            ),
+            id="operator_new",
+        ),
+        pytest.param(
+            "	 5501 [unknown] (/bin/cat)",
+            dict(
+                dso_true="(/bin/cat)",
+                dso_false="(/bin/cat)",
+            ),
+            id="unknown_bin_cat",
+        ),
+        pytest.param(
+            "	 7fd4 [unknown] (/lib/libudev.so (deleted))",
+            dict(
+                dso_true="(/lib/libudev.so (deleted))",
+                dso_false="(/lib/libudev.so (deleted))",
+            ),
+            id="libudev_deleted",
+        ),
+        pytest.param(
+            "	 1c00 [unknown] ([unknown])",
+            dict(
+                dso_true="[unknown] (unknown)",
+                dso_false="[unknown]",
+            ),
+            id="uknown_unknown",
+        ),
+        pytest.param(
+            "	 7fdb [unknown] ([vdso])",
+            dict(
+                dso_true="(vdso)",
+                dso_false="(vdso)",
+            ),
+            id="unknown_vdso",
+        ),
+        pytest.param(
+            "	 7f51 __gettime+0x1 ([vdso])",
+            dict(
+                dso_true="__gettime (vdso)",
+                dso_false="__gettime",
+            ),
+            id="gettime_vdso",
+        ),
+        pytest.param(
+            "	 ffa5 dup_mm+0x3f5 ([kernel.kallsyms])",
+            dict(
+                dso_true="dup_mm_[k]",
+                dso_false="dup_mm_[k]",
+            ),
+            id="dup_mm_kernel",
+        ),
+        pytest.param(
+            "	 dbdb CancelableTask::Run()+0x3b (/root/node)",
+            dict(
+                dso_true="CancelableTask::Run() (/root/node)",
+                dso_false="CancelableTask::Run()",
+            ),
+            id="cancelable_task_run_node",
+        ),
+        pytest.param(
+            "	 4090 recursive+0x1e (/native)",
+            dict(
+                dso_true="recursive (/native)",
+                dso_false="recursive",
+            ),
+            id="recursive_native",
+        ),
+        pytest.param(
+            "	 7fa0 @plt+0x0 (/lib/libc.so)",
+            dict(
+                dso_true="@plt (/lib/libc.so)",
+                dso_false="@plt",
+            ),
+            id="plt_libc",
+        ),
+        pytest.param(
+            "	ff84 page_fault+0x34 ([kernel.kallsyms])\n"
+            "	7f6e [unknown] (/tmp/perf-123.map)\n"
```

Comment:
How did `[unknown] (/tmp/perf-123.map)` look like before this change? (in the non-DSO mode)

Reply:
(I meant as with https://github.com/Granulate/gprofiler/pull/491#discussion_r995650623, how did it look after the `_collapse_stack` function)

So it looked like:
```
In [6]: _collapse_stack("aaa", "7f6e [unknown] (/tmp/perf-123.map)")                                                                                                                                       
Out[6]: 'aaa;[/tmp/perf-123.map]'
```
and now in both DSO and non-DSO cases it will be `(/tmp/perf-123.map)` which is good :)